force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,9 +180,21 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
+        
+        Account getAccount = new Account(Name = accountName);
+        upsert getAccount;
+
+        List<Opportunity> upsertOppsList = new List<Opportunity>(); 
+
         // Loop through each Opportunity name
             // Create an Opportunity with the given name
             // Add the Opportunity to the list
+        for(String s : oppNames){
+            upsertOppsList.add(new Opportunity(Name = s, AccountId = getAccount.Id, StageName = 'New', CloseDate = System.today()));
+        }
+
+        upsert upsertOppsList;
```

Comment:
This method, the way it's written, should have failed the test method. Here we need to (1) check to see if the Account already exists in the system (if so, grab it's ID, if not -- create it), and (2) check to see if any of the Opportunities already exist in the System (if so, update them, if not -- create them). When you find the time, re-write this code so that it passes the test method. Let me know if you have questions or need help!

Question 9 should give you an idea on how to handle the Account (and Opportunities) to some degree.